src/python/merge_statistics.py

```diff
@@ -0,0 +1,204 @@
+    statistics["imports"] = are_imports_involved
+
+    # Check if non-java files are involved.
+    non_java_files = diff_contains_non_java_file(repo, left_sha, right_sha)
+    if non_java_files is None:
+        raise ValueError("Could not compute if non-Java files are involved.")
+    statistics["non_java_files"] = non_java_files
+
+    # Return the row.
+    return pd.DataFrame([statistics])
+
+
+if __name__ == "__main__":
+    # Create CLI arguments.
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--merges_csv",
+        help="CSV file with merges that have been tested",
+        type=str,
+        default="results/combined/result.csv",
+    )
+    parser.add_argument(
+        "--output_dir",
+        help="Output directory for the statistics",
+        type=str,
+        default="results/combined",
+    )
+    args = parser.parse_args()
+
+    # Load the CSV file.
+    data = pd.read_csv(args.merges_csv, index_col="idx")
+
+    # Create result dataframe.
+    results = pd.DataFrame(
+        columns=[
+            "idx",
+            "num_files",
+            "num_intersecting_files",
+            "num_hunks",
+            "num_lines",
+            "num_intersecting_lines",
+            "imports",
+            "non_java_files",
+        ]
+    )
+
+    # Create failed results dataframe.
+    failed_results = pd.DataFrame(
+        columns=[
+            "idx",
+            "num_files",
+            "num_intersecting_files",
+            "num_hunks",
+            "num_lines",
+            "num_intersecting_lines",
+            "imports",
+            "non_java_files",
+        ]
+    )
+
+    with Progress(
+        SpinnerColumn(),
+        TextColumn("[progress.description]{task.description}"),
+        BarColumn(),
+        TimeElapsedColumn(),
+        TimeRemainingColumn(),
+    ) as progress:
+        task = progress.add_task(
+            f"Computing statistics for {data.shape[0]} merges", total=data.shape[0]
+        )
+
+        # Loop through each merge.
+        for idx, row in data.iterrows():
+            # Get data for a merge.
+            repo_slug = row["repository"]
+            left_sha = row["left"]
+            right_sha = row["right"]
+
+            # Compute statistics for a merge.
+            try:
+                statistics = compute_statistics(
+                    str(idx), repo_slug, left_sha, right_sha
+                )
+                results = pd.concat([results, statistics], ignore_index=True)
+            except Exception as e:
+                failed_results = pd.concat([failed_results, row], ignore_index=True)
+                print(f"Failed to compute statistics for {idx}: {e}")
+
+            # Update progress.
+            progress.update(task, advance=1)
+
+    # Save the results.
+    results.to_csv(f"{args.output_dir}/statistics.csv", index=False)
+    failed_results.to_csv(f"{args.output_dir}/failed_statistics.csv", index=False)
```

Comment:
If failed_results is non-empty, I think the script should exit with non-zero exit status.